files API doc

Aquassaut opened the pull request:
I still need to proofread this

Commit:
Added files API doc

Commit:
No more ad hoc json object

meganelacheny commented on source/includes/_files.md:
```diff
@@ -0,0 +1,194 @@
+Attribute | Description
+--------- | -----------
+`file_id` <br> *string* | Unique identifier for the file
+`url` <br> *string* | Generated URL for use withing the platform
```
"Generated URL for/to? use **within** the platform"

Aquassaut replied:
:+1:

meganelacheny commented on source/includes/_files.md:
```diff
@@ -0,0 +1,194 @@
+    {...},
+    {
+        "file_id": "cheese_data.csv",
+        "uri": "odsfile://cheese_data.csv",
```
"ur**l**": "odsfile://cheese_data.csv"

Aquassaut replied:
:+1:

meganelacheny commented on source/includes/_files.md:
````diff
@@ -0,0 +1,194 @@
+```json
+{
+    "file_id": "cheese_data.csv",
+    "uri": "odsfile://cheese_data.csv",
````
"ur**l**": "odsfile://cheese_data.csv"

Aquassaut replied:
:+1:

meganelacheny commented on source/includes/_files.md:
````diff
@@ -0,0 +1,194 @@
+POST https://{DOMAIN_ID}.opendatasoft.com/api/management/v2/files/
+```
+
+Uploads a file to the platform.
````
You can delete this

Aquassaut replied:
:+1:

meganelacheny commented on source/includes/_files.md:
````diff
@@ -0,0 +1,194 @@
+}
+```
+
+On success, a HTTP 200 is returned along with the file object of the uploaded file.
````
"a**n** HTTP 200"

meganelacheny commented on source/includes/_files.md:
```diff
@@ -0,0 +1,194 @@
+
+## Download a file
+
+This endpoint is for downloading a file from the platform.
```
"This endpoint is for downloading a file from the platform, **with the provided file_id**."

Aquassaut replied:
:ok_hand:

meganelacheny commented on source/includes/_files.md:
````diff
@@ -0,0 +1,194 @@
+GET https://{DOMAIN_ID}.opendatasoft.com/api/management/v2/files/{FILE_ID}/
+```
+
+Downloads the file with the provided file_id from the platform.
````
You can also delete this

Aquassaut replied:
:ok:

meganelacheny commented on source/includes/_files.md:
````diff
@@ -0,0 +1,194 @@
+Peugeot,white
+```
+
+On success, a HTTP 200 is returned with the file as an attachment.
````
"a**n** HTTP 200"

Aquassaut replied:
I don't care either way, but it seems that more people agree with me than with you :smile:

![](http://i.imgur.com/pSJ4xgw.png)
![](http://i.imgur.com/pxvn6BF.png)

Aquassaut replied:
Although literary references [seem to agree with you](https://books.google.com/ngrams/graph?content=a+http%2Can+http&year_start=1800&year_end=2008&corpus=15&smoothing=10&share=&direct_url=t1%3B%2Ca%20http%3B%2Cc0%3B.t1%3B%2Can%20http%3B%2Cc0)

meganelacheny commented on source/includes/_files.md:
````diff
@@ -0,0 +1,194 @@
+POST https://{DOMAIN_ID}.opendatasoft.com/api/management/v2/files/pasted_text/
+```
+
+Create a new file from textual data passed in the request body.
````
You can also delete this

Aquassaut replied:
This section will be removed

meganelacheny commented on source/includes/_files.md:
````diff
@@ -0,0 +1,194 @@
+}
+```
+
+On success, a HTTP 200 is returned with the file object of the newly created file.
````
"a**n** HTTP 200"

Commit:
Integrate PR feedbacks

Fixes some bad wording, missing words, removing redundancies

Commit:
Update doc for files API update

Removal of /files/pastedtext/, change of the download to /files/<file_id>/download,
addition to /files/<file_id>/ to list one file, addition of a new way of
sending file content to POST /files/

remilejeune commented on source/includes/_files.md:
```diff
@@ -0,0 +1,199 @@
+`file_id` <br> *string* | Unique identifier for the file
+`url` <br> *string* | Generated URL for use within the platform
+`filename` <br> *string* | Human readable name of the file
+`properties` <br> *object* | Set of file properties, such as its mimetype
```
Should be defined, same as the params json object in the records_errors PR, we need to explain what this object is.

Aquassaut replied:
Sure, but really, I have no idea what they can be except for mimetype...

remilejeune commented on source/includes/_files.md:
```diff
@@ -0,0 +1,199 @@
+`url` <br> *string* | Generated URL for use within the platform
+`filename` <br> *string* | Human readable name of the file
+`properties` <br> *object* | Set of file properties, such as its mimetype
+`created` <br> *string* | Time at which the change was made
```
created/change ?
Time at which the file was uploaded/created ?

Aquassaut replied:
Looks like a bad copy paste :-)

remilejeune commented on source/includes/_files.md:
```diff
@@ -0,0 +1,199 @@
+## List all files
+
+This endpoint lists all files available to the user who performs the request.
+
```
We need to explain the rules.
Why I see a file upload why somebody else but not all the files (edit_dataset right).

Aquassaut replied:
:+1:

remilejeune commented on source/includes/_files.md:
```diff
@@ -0,0 +1,199 @@
+
+## List one file
+
+This endpoints lists the file with provided file_id.
```
-lists +details/explains ?

Aquassaut replied:
Well, it doesn't explain or detail... I guess "retrieve" would work

remilejeune commented on source/includes/_files.md:
```diff
@@ -0,0 +1,199 @@
+# Files
+
+On the OpenDataSoft platform, files can be used as data source to create new datasets.
```
It can be used as attachment.
Maybe more context, like: to create a dataset from a file, the first thing to do is to upload the file to the platform ? @meganelacheny what do you think ? Is it useful ?

Aquassaut replied:
I didn't have the attachment part in mind, I should add it.

remilejeune commented on source/includes/_files.md:
````diff
@@ -0,0 +1,199 @@
+
+```json
+{
+    "file_id": "cheese_data.csv",
````
Don't we want to use file_uid ? (only in the API no migration needed).
To be consistent with processor_uid, dataset_uid, change_uid.
@jmorel  ?

Aquassaut replied:
We talked about it together and we decided that since it doesn't fit the general UID pattern, it should be left as ID. Open to changing it if you changed your mind though.

remilejeune commented on source/includes/_files.md:
````diff
@@ -0,0 +1,199 @@
+POST https://{DOMAIN_ID}.opendatasoft.com/api/management/v2/files/
+```
+
+This endpoint can receive a file sent in multipart, or file content sent through the content POST parameter, along with an optional mimetype and filename.
````
Explain that the classic multipart behavior is the html form post.

Aquassaut replied:
:+1:

remilejeune commented:
BTW, there is a conflict, and no merge from develop since a long time ;)

remilejeune commented on source/includes/_files.md:
```diff
@@ -0,0 +1,199 @@
+Attribute | Description
+--------- | -----------
+`file_id` <br> *string* | Unique identifier for the file
+`url` <br> *string* | Generated URL for use within the platform
```
Do we want to explain that the url will always start with odsfile:// ?

Aquassaut replied:
why not

Commit:
Updated doc with feedbacks

* Fix some errors (bad copy paste, double multipart instead of POST...)
* Removed optionality in mimetype
* Updated example POST upload for json payload
* Better wording
* Updated descriptions
* Added example multipart HTML form

Commit:
Updated file url scheme description

Commit:
Merge remote-tracking branch 'origin/develop' into feature/files

Commit:
Fix URL form for file download

meganelacheny commented on source/includes/_files.md:
```diff
@@ -66,7 +66,7 @@ GET https://{DOMAIN_ID}.opendatasoft.com/api/management/v2/files/
 ## List one file
 
-This endpoints lists the file with provided file_id.
+This endpoints is for retrieving the file object with provided file_id.
```
"This **endpoint** is (...)"

Commit:
Fix typo

jmorel commented:
* [x] Add paragraph about the way we expect parameters to be passed (for GET, PUT, POST and DELETE)
* [x] Add paragraph about multipart
* [x] Add paragraph about creating files from text (JSON)

Commit:
Merge remote-tracking branch 'origin/develop' into feature/files

Commit:
Add paragraph about the way we expect parameters to be passed

Commit:
Adds a paragraph with method of sending files

remilejeune commented on source/includes/_files.md:
````diff
@@ -0,0 +1,208 @@
+> Definition
+
+```HTTP
+GET https://{DOMAIN_ID}.opendatasoft.com/api/management/v2/files/download_file/{FILE_ID}
````
Documentation does not match the code.
In the code, the url is: /api/management/v2/download_file/{FILE_ID} (without the files)

Aquassaut replied:
Ah, yes :grimacing: 

:+1:

remilejeune commented on source/includes/_files.md:
```diff
@@ -0,0 +1,208 @@
+
+This endpoint can receive a file sent in multipart, or file content sent through the content POST parameter, along with a mimetype and an optional filename.
+
+### Multipart file upload
```
I think that we want to clarify the 240 Mo limit.

Aquassaut replied:
:+1: makes sense

Aquassaut replied:
(done in bb3a102)

remilejeune commented on source/includes/_files.md:
````diff
@@ -0,0 +1,208 @@
+
+```shell
+curl -XPOST 'https://yourdomain.opendatasoft.com/api/management/v2/files/' \
+    -d '{"content": "language,phrase\nEnglish,Hello World\Esperanto,Saluton mondo\n", "mimetype": "text/csv", "filename": "data.csv"}
````
One ' is missing

remilejeune replied:
And it returns an error:

```
{
  "status_code": 400,
  "message": "No file, or content from which to create a file was uploaded",
  "raw_params": {},
  "raw_message": "No file, or content from which to create a file was uploaded",
  "error_key": "MissingFileOrContentException"
```

Commit:
Fix small files doc problems (list below)

* Talk about the 240Mb limit
* Fix the file download URL
* Add the content-type header for file creation by content

remilejeune commented on source/includes/_files.md:
````diff
@@ -0,0 +1,209 @@
+]
+```
+
+## List one file
````
Should be something like 'Retrieve one file informations'

Aquassaut replied:
:+1:

Commit:
Rename the single file list chapter

Aquassaut commented:
Any news on this?
